egs/wsj/s5/steps/cleanup/create_segments_from_ctm.pl

```diff
@@ -325,7 +325,7 @@ sub ProcessWav {
   # Save the aligned CTM if needed
   if(defined($ACT)){
     for (my $i=0; $i<=$#aligned_ctm; $i++) {
```

Comment:
Sorry just saw this. That part of the code was added by Nagendra and he
originally just wanted to dump some temporary file instead of a real CTM I
think. And yes the script uses Kaldi format CTM which uses recording-id
instead of wav-id (it creates a segment file which uses recording-id's). I
can make a pull request to change the wav_id to recording_id if that's not
already done.

Guoguo

On Mon, Apr 4, 2016 at 1:18 PM, Daniel Povey <notifications@github.com>
wrote:

> In egs/wsj/s5/steps/cleanup/create_segments_from_ctm.pl
> <https://github.com/kaldi-asr/kaldi/pull/651#discussion_r58441040>:
>
> > @@ -325,7 +325,7 @@ sub ProcessWav {
> >    # Save the aligned CTM if needed
> >    if(defined($ACT)){
> >      for (my $i=0; $i<=$#aligned_ctm; $i++) {
>
> If it's called CTM it should be a correctly formatted CTM. It seems the
> script requires the input to be one of those Kaldi-style CTMs where instead
> of "wav-id channel" it has "recording-id 1", or "recording-id A" (they're
> equivalent). This should be documented at the top of the script. It can
> just print out another Kaldi-style ctm, with "recording_id 1" at the start,
> but the wav_id in the script should be changed to say recording_id, because
> that's what it really is. Remi, could you modify the pull request like
> this? Cc'ing Nagendra in case he was relying on this output for something.
> Dan
> … <#m_-177697443212186889_>
> On Mon, Apr 4, 2016 at 3:26 PM, Rémi Francis <notifications@github.com>
> wrote: In egs/wsj/s5/steps/cleanup/create_segments_from_ctm.pl <#651
> (comment)
> <https://github.com/kaldi-asr/kaldi/pull/651#discussion_r58432923>>: > @@
> -325,7 +325,7 @@ sub ProcessWav { > # Save the aligned CTM if needed >
> if(defined($ACT)){ > for (my $i=0; $i<=$#aligned_ctm; $i++) { It currently
> prints " ", where error is 1 if the word was right in the decoded ctm and 0
> otherwise (or the opposite, can't remember). I guess it would be nicer to
> have this a correctly formatted CTM. I just needed the wav id, so I didn't
> bother making it right. — You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/651/files/c90a6034fdf2881f97beaaabab1dd32766125d2c#r58432923>
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/651/files/c90a6034fdf2881f97beaaabab1dd32766125d2c#r58441040>
>